### PR TITLE
Clarify data not uploaded

### LINKED ISSUE #1137: Make it extra-clear to users that data is never uploaded
Users are still getting confused about whether their confidential data is being uploaded online. We should make it unambiguous and intuitive that data always stay local and is only read by the browser when training data locally. Some ideas:
- Instant feedback: The moment files are selected, show parsed metadata immediately: row counts, column names, file sizes, a preview. Anything that appears faster than an upload could possibly complete: "Read 48,210 rows locally in 0.3s."
- Live data transfer counter: "Data uploaded: 0 bytes." Then, when training rounds send gradients, the counter shows exactly what left: "Model updates sent: 2.1 MB (weights only — no raw data)."
- data stays local icon with secure icon
- suggest to users to test it by turning off their wifi when connecting the data (though it doesn't prove that data is not uploaded later during training) or checking the devtools network tab

Mock up:

Image

### PR DESCRIPTION
**Pull request by hleepfl**
# Changes
- Move lock icon around and make it more visible
- Remove as many visual elements as possible to give focus to the padlock
- Add a folder icon for local file selection. Probably doesn't do much ? idk
- List files when uploaded

(will fix e2e once design is finalized)

# Screenshots
Before
image
image
After
image
image

Hopefully closes #1137

### CODE REVIEW

File: webapp/src/components/dataset_input/FileSelection.vue [hunk 1]
```diff
@@ -8,6 +8,17 @@
   >
     <section>
       <!-- Hide the file input field when already submitted-->
+      <div v-if="noUpload" class="flex justify-center items-center my-1">
+        <span
+          v-tippy="{
+            content: 'Data always stays on your device and is never shared.',
+          }"
+          class="bg-green-200 text-green-500 rounded-full hover:cursor-pointer px-2 py-1"
```
Comment (JulienVig) [inline webapp/src/components/dataset_input/FileSelection.vue]: I think the colors are a bit off:
- Mallo recommended [this website](https://color.review/) to check the legibility and the current combination fails to pass
- It looks too bright in the dark theme, the bg and text colors should be swapped I believe
- I find the colors in the mockups of #1137 look quite nice

File: webapp/src/components/dataset_input/FileSelection.vue [hunk 2]
```diff
@@ -8,6 +8,17 @@
   >
     <section>
       <!-- Hide the file input field when already submitted-->
+      <div v-if="noUpload" class="flex justify-center items-center my-1">
+        <span
+          v-tippy="{
+            content: 'Data always stays on your device and is never shared.',
```
Comment (JulienVig) [inline webapp/src/components/dataset_input/FileSelection.vue]: I think this tooltip doesn't add much to the button text, what do you think of the mockup text: "Your files are read locally in your browser. They are never to our servers or anyone else." ?

File: webapp/src/components/dataset_input/FileSelection.vue [hunk 3]
```diff
@@ -21,13 +32,14 @@
         <p
           class="p-4 text-lg text-disco-blue dark:text-disco-light-blue flex-wrap justify-center"
```
Comment (JulienVig) [inline webapp/src/components/dataset_input/FileSelection.vue]:
```suggestion
          class="p-4 text-lg text-disco-blue dark:text-white flex-wrap justify-center"
```
While we're on this part of the code, let's make the dark mode text white like the rest of the page. The light blue isn't very legible

File: webapp/src/components/dataset_input/FileSelection.vue [hunk 4]
```diff
@@ -8,6 +8,17 @@
   >
     <section>
       <!-- Hide the file input field when already submitted-->
+      <div v-if="noUpload" class="flex justify-center items-center my-1">
```
Comment (JulienVig) [inline webapp/src/components/dataset_input/FileSelection.vue]:
```suggestion
      <div v-if="noUpload" class="flex justify-center items-center my-2">
```
Looks a bit too close to me

**Review by JulienVig — APPROVED** [review]
A few suggestions but otherwise all good for me! Most important one is the bg and text color for the secure icon text.
Are we giving up the "try with wifi off" idea?

File: webapp/src/components/dataset_input/FileSelection.vue (hunk 1 above)
Comment (hleepfl) [reply webapp/src/components/dataset_input/FileSelection.vue]: Thanks for the link ! Colors are more sane now

File: webapp/src/components/dataset_input/FileSelection.vue (hunk 2 above)
Comment (hleepfl) [reply webapp/src/components/dataset_input/FileSelection.vue]: How about
"Your files are read locally in your browser, who can only read them if you manually select them first. They are never sent to our servers or anyone else\'s. To convince yourself, you can start a training session (without collaboration) while turning off your machine\'s wifi !"
?

Comment (JulienVig) [reply webapp/src/components/dataset_input/FileSelection.vue]: I find the "manually select" part potentially confusing, what do you think of "Your files are read locally in your browser, they are never sent to our servers or anyone else's. To convince yourself, you can start a local training while turning off your machine's Wi-Fi!"

File: webapp/src/components/dataset_input/FileSelection.vue (hunk 1 above)
Comment (JulienVig) [reply webapp/src/components/dataset_input/FileSelection.vue]: Looks great!

File: webapp/src/components/dataset_input/FileSelection.vue (hunk 2 above)
Comment (hleepfl) [reply webapp/src/components/dataset_input/FileSelection.vue]: Sold
